website/content/v1.7/talos-guides/install/cloud-platforms/aws.md

```diff
@@ -106,191 +156,196 @@ Once the `SnapshotTaskDetail.Status` indicates `completed`, we can register the
 #### Register the Image
 
 ```bash
-aws ec2 register-image \
-    --region $REGION \
-    --block-device-mappings "DeviceName=/dev/xvda,VirtualName=talos,Ebs={DeleteOnTermination=true,SnapshotId=$SNAPSHOT,VolumeSize=4,VolumeType=gp2}" \
+AMI=$(aws ec2 register-image \
+    --block-device-mappings "DeviceName=/dev/xvda,VirtualName=talos,Ebs={DeleteOnTermination=true,SnapshotId=$SNAPSHOT_ID,VolumeSize=4,VolumeType=gp2}" \
     --root-device-name /dev/xvda \
     --virtualization-type hvm \
     --architecture x86_64 \
     --ena-support \
-    --name talos-aws-tutorial-ami
+    --name talos-aws-tutorial-ami \
+    --query 'ImageId' \
+    --output text)
 ```
 
 We now have an AMI we can use to create our cluster.
-Save the AMI ID, as we will need it when we create EC2 instances.
-
-```bash
-AMI="(AMI ID of the register image command)"
-```
 
 ### Create a Security Group
 
 ```bash
-aws ec2 create-security-group \
-    --region $REGION \
+SECURITY_GROUP_ID=$(aws ec2 create-security-group \
+    --vpc-id $VPC_ID \
     --group-name talos-aws-tutorial-sg \
-    --description "Security Group for EC2 instances to allow ports required by Talos"
-
-SECURITY_GROUP="(security group id that is returned)"
+    --description "Security Group for EC2 instances to allow ports required by Talos" \
+    --query 'GroupId' \
+    --output text)
 ```
 
 Using the security group from above, allow all internal traffic within the same security group:
 
 ```bash
 aws ec2 authorize-security-group-ingress \
-    --region $REGION \
-    --group-name talos-aws-tutorial-sg \
+    --group-id $SECURITY_GROUP_ID \
     --protocol all \
     --port 0 \
-    --source-group talos-aws-tutorial-sg
+    --source-group $SECURITY_GROUP_ID
 ```
 
-and expose the Talos and Kubernetes APIs:
+Expose the Talos (50000-50001) and Kubernetes APIs (6443).
 
 ```bash
 aws ec2 authorize-security-group-ingress \
-    --region $REGION \
-    --group-name talos-aws-tutorial-sg \
-    --protocol tcp \
-    --port 6443 \
-    --cidr 0.0.0.0/0
-
-aws ec2 authorize-security-group-ingress \
-    --region $REGION \
-    --group-name talos-aws-tutorial-sg \
-    --protocol tcp \
-    --port 50000-50001 \
-    --cidr 0.0.0.0/0
+    --group-id $SECURITY_GROUP_ID \
+    --ip-permissions \
+        IpProtocol=tcp,FromPort=50000,ToPort=50001,IpRanges="[{CidrIp=0.0.0.0/0}]" \
```

Comment:
I'm confused with this rule. We should expose 50000 on the controlplanes to the world, but we should never expose 50001 to the world.

Reply:
I realized 6443 should only be exposed on the VPC too. I'll update it.

Reply:
FWIW the existing guide exposes 6443, 50000, and 50001 to 0.0.0.0/0

Reply:
50001 should be never exposed outside of the cluster.

50000 - we recommend only controlplanes
6443 - same only controlplanes (unless you use AWS LB), in that case you don't need to expose it?

Reply:
https://www.talos.dev/v1.7/learn-more/talos-network-connectivity/